src/Player.js

```diff
@@ -0,0 +1,393 @@
+        onBufferProgress(initialPosition, progress)
+      );
+
+      clip.once('ended', () => {
+        if (this.nextTrack) {
+          let nextTrack = this.nextTrack;
+          let currentTrack = this.currentlyPlaying?.track;
+          this.nextTrack = null;
+
+          this.playTrack(nextTrack);
+          this.onNextTrack(currentTrack, nextTrack);
+        } else {
+          this.stopAll();
+          this.onPlaybackEnded();
+        }
+        this.onPlaybackProgress(1);
+      });
+
+      clip.on('bufferchange', (isBuffering) => onBufferChange(isBuffering));
+
+      this._playbackPositionInterval = setInterval(() => {
+        const { duration, currentTime } = clip;
+        if (duration === 0 || duration < currentTime) return;
+        let progress = currentTime / duration;
+
+        if (progress < 0) {
+          progress = 0;
+        } else if (progress > 1) {
+          progress = 1; // Prevent playback progress from exceeding 1 (100%)
+        }
+
+        if (
+          !this.currentlyPlaying ||
+          progress < this.currentlyPlaying.lastReportedProgress // Prevent playback progress from going backwards
+        ) {
+          return;
+        }
+
+        this.currentlyPlaying.lastReportedProgress = progress;
+        this.onPlaybackProgress(progress);
+      }, 250);
+
+      return clip.play() || Promise.resolve();
+    } catch (err) {
+      this.onError(err);
+      return Promise.reject(err.toString());
+    }
+  }
+
+  _getClip(
+    url,
+    fileSize,
+    initialPosition = 0,
+    lastAllowedPosition = 1,
+    audioMetadata = {}
+  ) {
+    if (typeof url !== 'string') {
+      throw new ProtonPlayerError('Invalid URL');
+    }
+
+    if (typeof fileSize !== 'number') {
+      throw new ProtonPlayerError('Invalid file size');
+    }
+
+    if (this.clips[url]) {
+      return this.clips[url];
+    }
+
+    const clip = new Clip({
+      url,
+      fileSize,
+      initialPosition,
+      lastAllowedPosition,
+      audioMetadata,
+      silenceChunks: this.silenceClipState.chunks,
+      volume: this.volume,
+      osName: this.osName,
+      browserName: this.browserName,
+      useMediaSource: canUseMediaSource(),
+    });
+
+    clip.on('loaderror', (err) => {
+      error('Clip failed to load', err);
+    });
+
+    clip.on('playbackerror', (err) => {
+      error('Something went wrong during playback', err);
+    });
+
+    this.clips[url] = clip;
+    return clip;
+  }
+
+  stopAll() {
+    this.currentlyPlaying = null;
+    this._clearIntervals();
+    Object.keys(this.clips).forEach((k) => {
+      this.clips[k].offAll('loadprogress');
+      this.clips[k].stop();
+    });
+  }
+
+  resume() {
+    if (this.currentlyPlaying && this.currentlyPlaying.clip) {
+      this.currentlyPlaying.clip.resume();
+    }
+  }
+
+  pause() {
+    if (this.currentlyPlaying && this.currentlyPlaying.clip) {
+      this.currentlyPlaying.clip.pause();
+    }
```

Comment:
This could be simplified to:
```suggestion
    this.currentlyPlaying?.clip?.pause();
```